security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -355,10 +360,6 @@ func (s *sdsservice) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecre
 				return err
 			}
 
-			// Remove the secret from cache, otherwise refresh job will process this item(if envoy fails to reconnect)
-			// and cause some confusing logs like 'fails to notify because connection isn't found'.
-			defer s.st.DeleteSecret(conID, resourceName)
-
```

Comment:
If there are other s.st.DeleteSecret() calls, please replace with releaseResourcePerConn() so that it is guaranteed the connection and cache are both cleaned up.

Reply:
Ok